api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -467,6 +468,36 @@ message HttpConnectionManager {
   // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
   // `HTTP spec <https://tools.ietf.org/html/rfc3986>` and is provided for convenience.
   bool merge_slashes = 33;
+
+  // [#not-implemented-hide:]
+  // Configuration of local reply returned by Envoy. Allows to specify mappings and format of
+  // response.
+  LocalReplyConfig local_reply_config = 36;
+}
+
+message LocalReplyConfig {
+  // Configuration of list of mappers which allows to filter and change local response.
+  // The client will iterate through mappers until first match.
+  repeated ResponseMapper mapper = 1;
+
+  // Allows to define custom format of local reply.
+  // It is allowed to use :ref:`command operators <config_access_log_command_operators>`
+  // which will be resolved to actual data.
+  type.StringOrJson format = 2;
+}
+
+message ResponseMapper {
+  // Filter is used to determine if the response should be changed.
+  accesslog.v2.AccessLogFilter filter = 1 [(validate.rules).message = {required: true}];
+
+  // Rewriter defines which values in local reply should be changed.
+  ResponseRewriter rewriter = 2 [(validate.rules).message = {required: true}];
+}
+
+// Configuration of new value for matched local response.
+message ResponseRewriter {
```

Comment:
Should there be an ability to override header values?
For example to set content-type or some application-specific chillout header.

Reply:
It might be added later.

Reply:
Agree.